src/main/java/com/mercateo/sqs/utils/visibility/VisibilityTimeoutExtender.java

```diff
@@ -70,14 +70,17 @@ private Integer timeoutInSeconds(Duration timeout) {
     @Override
     public void run() {
         try {
-            log.trace("changing message visibility: " + request);
-            retryer.call(() -> sqsClient.changeMessageVisibility(request));
-        } catch (AmazonServiceException e) {
-            errorHandlingStrategy.handleExtendVisibilityTimeoutException(e, message);
+            retryer.call(() -> messageWrapper.changeMessageVisibility(sqsClient, request));
         } catch (Exception e) {
-            log.error("error while extending message visibility for " + message.getHeaders().get("MessageId",
-                    String.class), e);
-            throw new RuntimeException(e);
+            if (e.getCause() instanceof AlreadyAcknowledgedException) {
+                return;
```

Comment:
Why introducing the new exception if we just ignore it.

Reply:
ok, removed with https://github.com/Mercateo/sqs-utils/pull/23/commits/adda920c9cd24ca10b5462181294440ca9dc8f11